### ISSUE TITLE
Assignments and Layout/MultilineOperationIndentation

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#6108]
Hello!

I think we should change the default behavior of `Layout/MultilineOperationIndentation` cop, or make it more configurable.

I'm agree with this code:

```ruby
a = b +
  c +
  d
```

Even for `EnforcedStyle: indented`, I guess.

But I want to indent every line of assignment right part if it starts at the new line, with `indented` style:

```ruby
a =
  b +
    c +
      d
```

--------

## Expected behavior

No offences when assignment right part starts at the new line.

## Actual behavior

There are offenses.

## Steps to reproduce the problem

Code is above.

## RuboCop version

```
$ rubocop -V
0.58.1 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-linux)
```

**Comment by jaredbeck** [comment]
I'm opposed to changing the current definition of `EnforcedStyle: indented`. If we were to support the proposed style, I'd suggest introducing a new style name, perhaps `EnforcedStyle: indented_continuously`, or `indented_cumulatively`.

**Comment by AlexWayfer** [comment]
Another note:

If you have:

```ruby
some_model.some_method(some_argument) ||
  another_model.another_method(another_argument)
```

and you want assign in to variable, you should change indentation:

```ruby
result =
  some_model.some_method(some_argument) ||
  another_model.another_method(another_argument)
```

It looks strange for me.

But I'm not against the new style.

**Comment by JelF** [comment]
There is [test](https://github.com/rubocop-hq/rubocop/blob/613cc43/spec/rubocop/cop/layout/multiline_operation_indentation_spec.rb#L405-L412) that 

```ruby
  arg_array.size == a.size && (
    arg_array == a ||
    arg_array.map(&:children) == a.map(&:children)
 )
```

passes, but it cannot be enforced over


```ruby
  arg_array.size == a.size && (
    arg_array == a ||
      arg_array.map(&:children) == a.map(&:children)
 )
```

Can this also be added to cop?

**Comment by AlexWayfer** [comment]
And I've faced with this again, in another project.

```ruby
something =
  OneThing.where(...).all +
  AnotherThing.where(...).all
```
